syntax = "proto3";

package example;

service MyService {
  rpc SayHello (HelloRequest) returns (HelloReply);
  rpc ComputeStats (DataList) returns (StatsReply);
  rpc DetectAnomalies (DataList) returns (AnomalyReply);
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}


message AnomalyReply {
  repeated string anomalies = 1;
}

message Data {
  string id = 1;
  float temperatura = 2;
  float velocidade_vento = 3;
  float energia_prod = 4;
}

message DataList {
  repeated Data dados = 1;
}

message StatsReply {
  float media_temperatura = 1;
  float media_vento = 2;
  float media_energia = 3;
  string forecast = 4;
}